### PR TITLE
Update `PASSWORD_NAMES` to include `api_key` and remove `pass`

### PR DESCRIPTION
**Pull request by KittyBorgX**
This updates `PASSWORD_NAMES` in `crates/ruff/src/rules/flake8_bandit/helpers.rs` as per #2384

### CODE REVIEW

File: crates/ruff/src/rules/flake8_bandit/helpers.rs [hunk 1]
```diff
@@ -2,9 +2,7 @@ use rustpython_parser::ast::{Constant, Expr, ExprKind};
 
 use crate::checkers::ast::Checker;
 
-const PASSWORD_NAMES: [&str; 7] = [
-    "password", "pass", "passwd", "pwd", "secret", "token", "secrete",
-];
+const PASSWORD_NAMES: [&str; 6] = ["password", "passwd", "pwd", "secret", "token", "api_key"];
```
Comment (charliermarsh) [inline crates/ruff/src/rules/flake8_bandit/helpers.rs:5]: I'm wondering if, instead, we should consider using a regex to match Bandit's behavior?

https://github.com/PyCQA/bandit/blob/a9eaafa53ae73c1a66172373128cb19bb7b82a32/bandit/plugins/general_hardcoded_password.py#L13-L16

Comment (ngnpope) [reply crates/ruff/src/rules/flake8_bandit/helpers.rs:5]: Was going to say the same thing. This is a small improvement but doesn't solve the main problem I raised in the issue - that the current substring match causes matching parts of larger words.

Comment (KittyBorgX) [reply crates/ruff/src/rules/flake8_bandit/helpers.rs:5]: @charliermarsh would regex be in the scope of the linked issue? if so, i'll implement it in this pr.

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_bandit/helpers.rs:5]: Yeah I believe it's within scope.

**Comment by charliermarsh** [comment]
Closing as this was resolved by #3222.

**Pull request closed without merging**